### ISSUE TITLE
Incremental harvest query

### ISSUE THREAD

**Issue by telezoic** [DigitalNZ/supplejack_manager#157]
Hi folks, 

I'm hoping you could share with me the expected incremental harvest behaviour?

Does the 'incremental harvest' function only ingest records that are new to the target repo since the last harvest?

I've had what appear to be successful incremental harvests (that is, there are no errors) but all the  records from my target repo get harvested (and I can see that new records have been added to the repo). 

![incremental](https://user-images.githubusercontent.com/444648/81441247-4a539d80-9126-11ea-9e2f-755dcaa2ce77.png)

Does the incremental harvest depend on sidekiq running from the API directory, the way it's described for the full and flush harvest?

> Supplejack has it’s own instance of Sidekiq that you will need to run for the Full and Flush harvest to work. Start Sidekiq from the api directory with bundle exec sidekiq. 

> https://digitalnz.github.io/supplejack/components/supplejack-api.html

Any suggestions/guidance you could share would be greatly appreciated.

Thanks, 

Dan

**Comment by dan-dnz** [comment]
Hi Dan,

Apologies for the delayed response. 

**Expected incremental harvest behaviour:**
Running a harvest in 'incremental' mode means Supplejack will look for the date of the last successful harvest from the same parser. It will use this date to build the OAI query 
   `...?verb=ListRecords&from=<date>...`
This should mean that the OAI returns only those records that are new or have been updated since that date. This also includes deletions.

Some issues I have encountered in the past in this area are when the OAI returns no records since specified "from" date, the harvest runs correctly (eg doesnt harvest anything) but sometimes returns an error. Another difficulty I've had is that the generic OAI gem that Supplejack uses isn't very helpful in terms of logging so you cant really see what dates end up being used in the queries.

But your scenario seems more like something is going wrong where Supplejack is trying to find the date of the last successful harvest job. Its as if it doesnt find one, or doesnt pass the correct date on. So the harvest ends up running in full each time.

Unfortunately we've had some other important priorities at the end of this financial year and I haven't been able to get a dev to dig into this further for you. Although I did get a basic answer to your other question in that incremental harvests apparently **do not** depend on sidekiq running from the API directory.

Hopefully I've at least been a little bit helpful.

Let us know if you are still having issues or have any other questions.

Cheers
Dan
